crypto/src/merkle.rs

```diff
@@ -189,173 +196,134 @@ impl<T> Node<T> {
     }
 
     #[cfg(feature = "std")]
-    fn nodes_pair_hash(left: &Self, right: &Self) -> HashOf<Self> {
-        let left_hash = left.hash();
-        let right_hash = right.hash();
+    fn nodes_pair_hash(left: &Self, right: &Self) -> Option<HashOf<Self>> {
+        let (left_hash, right_hash) = match (left.hash(), right.hash()) {
+            (None, None) => return None,
+            (None, Some(r_hash)) => return Some(r_hash),
+            (Some(l_hash), None) => return Some(l_hash),
+            (Some(l_hash), Some(r_hash)) => (l_hash, r_hash),
+        };
         let sum: Vec<_> = left_hash
             .as_ref()
             .iter()
             .zip(right_hash.as_ref().iter())
             .map(|(l, r)| l.saturating_add(*r))
```

Comment:
why don't we use `wrapping_add`? Seems as if the `saturating_add` would hit a ceiling